feat(release): allow preserving zero-based versions

jsurkont opened the pull request:
closes #447

## Description

This change adds a customization for version bumping. Specifically, it allows changing the default version bumping rules in order to preserve zero-based releases.
 
<!--- Describe your changes in detail -->

## Motivation and Context

The current bumping rules are hard-coded, so it's necessary to extend the input arguments to allow for customization.

This PR adds a new CLI flag `--keep-zerover`. I also considered extending the configuration. However, it wasn't clear to me which section would be appropriate. Perhaps `changelog`? I think that having it in the configuration would be quite convenient to users. It would also mean that the signature of `Changelog::bump_version` doesn't need to be modified. What do you think?

This PR closes #447.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?

I extended the unit tests for `calculate_next_version`. I also run `git cliff` with and without the new flag.
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots / Logs (if applicable)

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

I wasn't sure if this should be considered a breaking change, but I've marked is a such to be on the safe side. My guess is that there wouldn't be a large number (if any) of users who use the current bump version feature with zero-based releases, as it would result in moving away from the zero-based system. 

- [ ] Bug fix (non-breaking change which fixesan issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other <!--- (provide information) -->

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

<!--- Thank you for contributing to git-cliff! ⛰️  -->

orhun commented:
I'm thinking if we should offer this as a config option (also environment variable) rather than a command-line argument. I don't think this will be used very often especially when the user is only interested in changelog generation rather than bumping the version. What do you think?

jsurkont commented:
I agree. I guess that, for the majority of projects, adding this option to the configuration (or using an environmental variable) would be the most convenient way.

Do you have a preference under which config section this option should be added?

orhun commented:
I think the best option would be to add a new section as follows:

```toml
[bump]
# Configures automatic minor version increments for feature changes.
# When true, a feature will always trigger a minor version update.
# When false, a feature will trigger:
# - a patch version update if the major version is 0.
# - a minor version update otherwise.
features_always_bump_minor = true
# Configures 0 -> 1 major version increments for breaking changes.
# When true, a breaking change commit will always trigger a major version update
# (including the transition from version 0 to 1)
# When false, a breaking change commit will trigger:
# - a minor version update if the major version is 0.
# - a major version update otherwise.
breaking_always_bump_major = true
```

What do you think?

jsurkont commented:
Yes, I think this is the cleanest solution ✨ 

I've just updated my PR to use the new config section. I also changed the code so it doesn't change the pubic api of `git-cliff-core` at a (small) cost of function repetition. Unfortunately, the "Semver" check is still failing because of adding a new field to the `Config` struct.

orhun commented on website/docs/configuration/bump.md:
Can we also mention the existence of this configuration option in `usage/bump-version.md`?

jsurkont replied:
Thanks for your feedback!

I added a new fixture and extended the `usage` docs so they now mention the new configuration options.

orhun replied:
Awesome, thanks!

orhun reviewed (commented):
Looks good! Can you also [add a test fixture](https://github.com/orhun/git-cliff/tree/main/.github/fixtures)? There is also one example under `test-bump-version`.

orhun reviewed (approved):
Awesome, thanks for your contribution!
